src/flamegraph/color/palettes.rs

```diff
@@ -99,6 +99,25 @@ pub(super) mod wakeup {
     }
 }
 
+pub(super) mod rust {
+    use crate::flamegraph::color::BasicPalette;
+
+    // Partition into user code and Rust system code
+    pub fn resolve(name: &str) -> BasicPalette {
+        if name.starts_with("core::") || name.starts_with("std::") {
+            return BasicPalette::Orange;
+        } else if name.contains("as core::")
+            || name.contains("as std::")
+            || name.contains("as alloc::")
```

Comment:
Do these show up in practice in debug symbols? If so, I think I'd prefer more robust parsing here, as I feel like this parsing might end up yielding false positives down the line. Can you narrow down exactly what kind of items we expect to hit this case?

Reply:
Yes, I saw things like the following in a local test:
```
<alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
<core::iter::adapters::enumerate::Enumerate<I> as core::iter::traits:iterator::Iterator>::fold
```
I'll search around for a more definitive answer, but these should probably match on something like `name.starts_with("<alloc::")`, etc.

Reply:
Yeah, I think `<alloc::` and `<core::` is probably a better match here then :+1:

Reply:
Sounds good.  I found [the name-mangling rfc](https://github.com/rust-lang/rfcs/blob/master/text/2603-rust-symbol-name-mangling-v0.md) where it says, "The RFC therefore proposes to keep symbol names close to how methods are represented in error messages".  My naive interpretation is that what we have here should work..